blocks/feed-newsroom/feed-newsroom.js

```diff
@@ -191,13 +196,33 @@ export default async function decorate(block) {
 </form>`;
   const uniqYears = Array.from(new Set(results.map((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return parseInt(itsDate[itsDate.length - 1], 10); })));
   // eslint-disable-next-line
-  const yroptions = uniqYears.reduce((accum, current) => { accum += "<option value='"+current+"'>" + current + "</option>"; return accum;},"");
-  filterDiv.querySelector('select').innerHTML = filterDiv.querySelector('select').innerHTML + yroptions;
+  const yroptions = uniqYears.reduce((accum, current) => { accum += "<option value='" + current + "'>" + current + "</option>"; return accum; }, "");
+  filterDiv.querySelector('#news_year').innerHTML = filterDiv.querySelector('#news_year').innerHTML + yroptions;
+  const categoryDetails = await fetchTagsOrCategories('', 'categories', 'newsroom', locale);
+  // eslint-disable-next-line
+  const categoryOptions = categoryDetails.reduce((accum, current) => { accum += "<option value='" + current.id + "'>" + current.id + "</option>"; return accum; }, "");
+  filterDiv.querySelector('#news_category').innerHTML = filterDiv.querySelector('#news_category').innerHTML + categoryOptions;
 
   filterDiv.querySelector('form .filter-nav button').addEventListener('click', () => {
-    const searchYear = Number(filterDiv.querySelector('form .filter-nav select').value);
-    searchResults = results.filter((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return (parseInt(itsDate[itsDate.length - 1], 10) === searchYear); });
-    loadYearResults(block, blockType, searchResults, blockCfg, locale);
+    const searchYear = Number(filterDiv.querySelector('form .filter-nav #news_year').value);
+    const searchCategory = filterDiv.querySelector('form .filter-nav #news_category').value;
+    if (!searchCategory && searchYear) {
+      const option = filterDiv.querySelector('#news_category').options;
+      for (let i = 0; i < option.length; i += 1) {
+        if (option[i].text === 'news') {
+          option[i].selected = true;
+          break;
+        }
+      }
+      searchResults = results.filter((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return (parseInt(itsDate[itsDate.length - 1], 10) === searchYear); });
```

Comment:
I think the logic can be simplified a bit here. The `queryObj` can be used to apply the `year` and `category` based `where` clauses. Something like..
```
if year is present
  queryObj.where((item) => {
    //logic for year based filtering
  });

if category is present
  queryObj.where((item) => {
    //logic for category based filtering
  });


```